src/components/TransactionHistoryRows/hooks.tsx

```diff
@@ -0,0 +1,74 @@
+import { HistoryTimeframe } from '@shapeshiftoss/types'
+import { Dex } from '@shapeshiftoss/unchained-client'
+import dayjs from 'dayjs'
+import { useEffect, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { TxDetails } from 'hooks/useTxDetails/useTxDetails'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectAssetById, selectCryptoPriceHistoryTimeframe } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { getTradeFees } from './utils'
+
+export const useTradeFees = ({ txDetails }: { txDetails: TxDetails }) => {
+  const [tradeFees, setTradeFees] = useState<string | null>(null)
+
+  const dispatch = useDispatch()
+  const cryptoPriceHistoryData = useAppSelector(state =>
+    selectCryptoPriceHistoryTimeframe(state, HistoryTimeframe.ALL),
+  )
+
+  const buyAsset = useAppSelector(state =>
+    selectAssetById(state, txDetails.buyAsset?.assetId ?? ''),
+  )
+  const sellAsset = useAppSelector(state =>
+    selectAssetById(state, txDetails.sellAsset?.assetId ?? ''),
+  )
+
+  const { findPriceHistoryByAssetId } = marketApi.endpoints
+
+  useEffect(() => {
+    if (
+      !(
+        txDetails?.tx?.trade &&
+        txDetails.sellAsset &&
+        txDetails.buyAsset &&
+        txDetails.sellTransfer &&
+        txDetails.buyTransfer
+      )
+    )
+      return
+    if (txDetails.tx.trade.dexName !== Dex.CowSwap) return
+
+    if (!Object.keys(cryptoPriceHistoryData).length) {
```

Comment:
Probably would make more sense to try accessing both AssetIds, which:
1. would be O(1) * 2 vs O(n)
2. might give us more confidence in the fact that these specific AssetIds are loaded

Reply:
https://github.com/shapeshift/web/pull/2357/commits/396b493807886a7acf0fd945ed621b5f6bd55989